pkg/operator/controller/externaldns/deployment_test.go

```diff
@@ -2817,6 +2817,8 @@ func TestDesiredExternalDNSDeployment(t *testing.T) {
 }
 
 func TestExternalDNSDeploymentChanged(t *testing.T) {
+	// TODO: refactor this test case to remove the mutation logic as it's misleading.
+	// This test case should have current/desired/expected fields.
```

Comment:
In my opinion, it makes the test cases easier to read when each test case only specifies what changes, which is what the `mutate` func shows.  I'm biased though.

Reply:
It is weird, though, that some test cases also specify the original deployment and the expected deployment.

Reply:
Yes, I think `mutate()` indeed may keep test cases shorter and clearer (clear what changed, I use test case name for this but it's less accurate). However I expected the mutate function to modify the desired (by the operator) state to get the current state while it's the opposite now.
The expected deployment may be different from the desired (e.g. unsolicited containers, volumes). I think it makes sense to verify what `externalDNSDeploymentChanged` sends to the API server at the end, that's why we need `expectedDeployment`.